### ISSUE TITLE
Fix S3427: Rule should not generated FPs for generic parameters

### ISSUE THREAD

**Issue by pms1969** [SonarSource/sonar-csharp#1964]
### Description

When running against a class with methods that differ by multiple generic params, a false positive is raised.

### Repro steps

Please provide the steps required to reproduce the problem

```csharp
public class MyClass {
    public static Property FromDto(this KeyValuePair<PropertyKey, CreatePropertyRequest> kvp, DateTimeOffset? effectiveDate = null){ ...}
    public static PerpetualProperty FromDto(this KeyValuePair<PropertyKey, CreatePerpetualPropertyRequest> kvp) {...}
}
```

In this case, the KVP's differ by the second generic type.
repro zip attached.
[repro.zip](https://github.com/SonarSource/sonar-csharp/files/2455184/repro.zip)

### Expected behavior

No issue raised.

### Actual behavior

Issue identified.

### Known workarounds

Other than silencing the issue, none.

### Related information

* SonarC# Version : 6.8.1
* Visual Studio Version N/A
* If running through the Scanner for MSBuild, its version 15.x

**Comment by Evangelink** [comment]
Hi @pms1969,

Thanks for the feedback! I have been able to reproduce the issue locally and it seems that we are not handling properly generic types.

**Comment by pms1969** [comment]
Ah, excellent.  Not just us then. :)
